app/test/unit/git/format-patch-test.ts

```diff
@@ -0,0 +1,39 @@
+import { setupTwoCommitRepo } from '../../helpers/repositories'
+import { Repository } from '../../../src/models/repository'
+import { formatPatch } from '../../../src/lib/git'
+import {
+  cloneLocalRepository,
+  makeCommit,
+} from '../../helpers/repository-scaffolding'
+import { GitProcess } from 'dugite'
+
+describe('formatPatch', () => {
+  describe('in a repo with commits', () => {
+    let repository: Repository
+    beforeEach(async () => {
+      repository = await setupTwoCommitRepo()
+    })
+    it('returns a string for a commit range', async () => {
+      expect(await formatPatch(repository, 'HEAD~', 'HEAD')).toBeTruthy()
+    })
+    it('returns empty string for no range', async () => {
+      expect(await formatPatch(repository, 'HEAD', 'HEAD')).toBeFalsy()
+    })
+    describe('applied in a related repo', () => {
+      let clonedRepository: Repository
+      beforeEach(async () => {
+        clonedRepository = await cloneLocalRepository(repository)
+        await makeCommit(clonedRepository, {
+          entries: [{ path: 'okay-file', contents: 'okay' }],
+        })
+      })
+      it('is valid', async () => {
+        const patch = await formatPatch(repository, 'HEAD~', 'HEAD')
```

Comment:
Have you tried testing with a range of multiple commits (`HEAD~..HEAD` is a single commit)?

I'm curious whether `git diff` as it's currently used will cover all the cases that `git format-patch` supports.

Reply:
yeah i think you're suspicion is right. `diff` will create a single patch instead of a series of patches.

i made this change due to limitations of buffering output from a child process. if there were too many commits, the output would be truncated. i experimented with raising the output buffer limit, but i had to make it very large to accommodate some test cases i tried, which resulted in tripling desktop's usual heap size.

i'm going to investigate some more alternatives. piping output directly from one child process to another doesn't get us around the buffering issue. so we could
* write a potentially large patch file to disk and read it back in (probably not as performant)
* write a script/command that pipes the output of one git command to another thereby avoiding the buffering issue (might have cross-platform issues)
* abandon the patch approach and simply execute the expected rebase command in the working tree and check the output for conflicts (probably not as performant)

Reply:
What about our current `spawn` APIs? This might be a bit more flexible for this use case:

https://github.com/desktop/desktop/blob/e71e497e7366f53820f408aa1bc49a7eb8ac6426/app/src/lib/git/spawn.ts#L13-L26

Reply:
thanks @shiftkey! that did the trick! (d828453cf90b950c0289942a25d427e60bb00f87)